ui/imports/shared/views/chat/LinksMessageView.qml

```diff
@@ -71,35 +71,39 @@ Column {
                 id: linkFetchConnections
                 enabled: false
                 target: root.messageStore.messageModule
-                onLinkPreviewDataWasReceived: {
-                    let response = {}
-                    try {
-                        response = JSON.parse(previewData)
-                    } catch (e) {
-                        console.error(previewData, e)
-                        linkMessageLoader.loadingFailed = true
-                        return
-                    }
-                    if (response.uuid !== linkMessageLoader.uuid) return
-                    linkFetchConnections.enabled = false
-
-                    if (!response.success) {
-                        console.error("could not get preview data")
-                        linkMessageLoader.loadingFailed = true
-                        return
-                    }
-
-                    linkData = response.result
-                    linkMessageLoader.loadingFailed = false
-
-                    linkMessageLoader.height = undefined // Reset height so it's not 0
-                    if (linkData.contentType.startsWith("image/")) {
-                        return linkMessageLoader.sourceComponent = unfurledImageComponent
-                    }
-                    if (linkData.site && linkData.title) {
-                        linkData.address = link
-                        return linkMessageLoader.sourceComponent = unfurledLinkComponent
-                    }
+                function onLinkPreviewDataWasReceived(previewData) {
+                    Qt.callLater((previewData) => {
```

Comment:
It deserves some comment in the code.

Also, to clarify, it doesn't release the main thread, it just puts the task at the end of the event queue, meaning it may be executed during scrolling anyway, it depends if QML engine returns to the event loop in the meantime.

Reply:
I've removed the Qt.callLater as it doesn't behave nicely with fast scrolling